Library/Homebrew/formula_auditor.rb

```diff
@@ -590,6 +590,29 @@ def audit_bottle_spec
       new_formula_problem "New formulae in homebrew/core should not have a `bottle do` block"
     end
 
+    def audit_eol
+      return unless @core_tap
+      return if formula.deprecated? || formula.disabled?
+
+      name = if formula.versioned_formula?
+        formula.name.split("@").first
+      else
+        formula.name
+      end
+
+      metadata = SharedAudits.eol_data(name, formula.version.major)
+      metadata ||= SharedAudits.eol_data(name, formula.version.major_minor)
+
+      return if metadata.blank? || metadata["eol"] == false
+
+      if metadata["eol"] == true
+        problem "Product is EOL"
```

Comment:
Is there a URL we can supply for this to the endoflife.date page?

Reply:
Something like this?
```suggestion
        problem "Product is EOL, see https://endoflife.date/#{name}"
```